nltk/draw/tsne.py

```diff
@@ -0,0 +1,173 @@
+#
+#  tsne.py
+#  
+# Implementation of t-SNE in Python. The implementation was tested on Python 2.5.1, and it requires a working 
+# installation of NumPy. In order to plot the results of this example, a working installation of matplotlib is required.
+# The example can be run by executing: ipython tsne.py -pylab
+#
+#
+#  Created by Laurens van der Maaten on 20-12-08.
+#  Copyright (c) 2008 Tilburg University. All rights reserved.
```

Comment:
Another option is to use http://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html 
It is an extra dependency, but the advantage is that we won't be adding more code to NLTK, and that we'll be teaching users to use scikit-learn - it is a good instrument to have in a toolbelt, and it is already NLTK dependency. 

I'm not quite comfortable with adding non-trivial math-heavy code with zero unit tests to NLTK; this module also doesn't follow pep8, and it is not clear it works in Python 3. Its license is also unclear.